setup.cfg

```diff
@@ -24,6 +24,7 @@ install_requires =
     yt-dlp>=2021.12.27
     pdfkit>=0.6.0
     tqdm>=4.0.0
+    lxml>=5.0.0
```

Comment:
What is lxml needed for? (I know it can be used by bs4 but I do not know why it is necessary)

Reply:
Well the response of /public.php/webdav is a xml-file which describes the structure of the sharing link we found.

To be able to parse xml we need to install lxml so that bs4 can use it.

Reply:
As that XML is highly structured it should be sufficient to take the XML parser from the stdlib

Reply:
I want to note that lxml is a more efficient parser than the standard bs4 parsers. You could get a speedup by replacing all html parsers with lxml in syncmymoodle.

see:
https://thehftguy.com/2020/07/28/making-beautifulsoup-parsing-10-times-faster/ (although maybe not 10x, could still get a significant speedup)

Reply:
Tried replacing all parsers with lxml and this is the result
(testcase was checking a single already downloaded course)

Using current main SMM: 13.159s
Using lxml:                           12.974s
Using lxml and cchardet:  12.189s

This was tested on a higher end PC, not sure how this looks like on other platforms.
Probably also highly fluctuates depending on the course you check against.

Reply:
Running it again a few times, and the runtime fluctuates a lot.  Speed up seems to be marginal (<10%).